diff -uNr sos-3.2/sos/policies/redhat.py sos-3.2p/sos/policies/redhat.py
--- sos-3.2/sos/policies/redhat.py	2017-05-19 14:48:57.254000005 +0200
+++ sos-3.2p/sos/policies/redhat.py	2017-05-19 14:50:33.871000506 +0200
@@ -34,9 +34,9 @@
 
 
 class RedHatPolicy(LinuxPolicy):
-    distro = "Red Hat"
-    vendor = "Red Hat"
-    vendor_url = "http://www.redhat.com/"
+    distro = "EuroLinux"
+    vendor = "EuroLinux"
+    vendor_url = "https://www.euro-linux.com/"
     _tmp_dir = "/tmp"
 
     def __init__(self):
@@ -102,9 +102,9 @@
 
 
 class RHELPolicy(RedHatPolicy):
-    distro = "Red Hat Enterprise Linux"
-    vendor = "Red Hat"
-    vendor_url = "https://access.redhat.com/support/"
+    distro = "EuroLinux"
+    vendor = "EuroLinux"
+    vendor_url = "https://support.euro-linux.com"
     msg = _("""\
 This command will collect diagnostic and configuration \
 information from this %(distro)s system and installed \
