From 5a97e0e2571b948f1f7bc602e6f190976de99eee Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Tue, 13 Jan 2015 16:31:58 +0000
Subject: [PATCH] [anaconda] make useradd password regex tolerant of whitespace

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/anaconda.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sos/plugins/anaconda.py b/sos/plugins/anaconda.py
index 08d300b..aea3300 100644
--- a/sos/plugins/anaconda.py
+++ b/sos/plugins/anaconda.py
@@ -54,7 +54,7 @@ class Anaconda(Plugin, RedHatPlugin):
         )
         self.do_file_sub(
             "/root/anaconda-ks.cfg",
-            r"(user.*--password=*)(\S*)",
+            r"(user.*--password=*\s*)\s*(\S*)",
             r"\1********"
         )
 
-- 
1.9.3

