diff -up sos-3.2/sos/plugins/openshift.py.org sos-3.2/sos/plugins/openshift.py
--- sos-3.2/sos/plugins/openshift.py.org	2016-02-01 15:36:40.094862160 -0500
+++ sos-3.2/sos/plugins/openshift.py	2016-02-01 15:37:35.420431535 -0500
@@ -147,5 +147,10 @@ class Openshift(Plugin, RedHatPlugin):
         self.do_file_sub(plugin_dir + 'openshift-origin-dns-nsupdate.conf',
                          r"(BIND_KEYVALUE\s*=\s*)(.*)",
                          r"\1********")
+        # LDAP authentication:  AuthLDAPBindPassword "IShouldNotBeHere"
+        ldap_paths = '/var/www/openshift/(broker|console)/httpd/conf.d/.*'
+        self.do_path_regex_sub(ldap_paths,
+                               r"(AuthLDAPBindPassword)\s*(.*)",
+                               r"\1********")
 
 # vim: et ts=4 sw=4
