From 0aec209ec6870c6ec3cbb6e2209d35adb2b62009 Mon Sep 17 00:00:00 2001
From: Sandro Bonazzola <sbonazzo redhat com>
Date: Tue, 16 Dec 2014 12:29:57 +0000
Subject: [PATCH] [ovirt] remove ovirt-engine setup answer file password leak

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>
Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/ovirt.py | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/sos/plugins/ovirt.py b/sos/plugins/ovirt.py
index b169b90..407c918 100644
--- a/sos/plugins/ovirt.py
+++ b/sos/plugins/ovirt.py
@@ -171,4 +171,19 @@ class Ovirt(Plugin, RedHatPlugin):
                 r'{key}=********'.format(key=key)
             )
 
+        # Answer files contain passwords
+        for key in (
+            'OVESETUP_CONFIG/adminPassword',
+            'OVESETUP_CONFIG/remoteEngineHostRootPassword',
+            'OVESETUP_DWH_DB/password',
+            'OVESETUP_DB/password',
+            'OVESETUP_REPORTS_CONFIG/adminPassword',
+            'OVESETUP_REPORTS_DB/password',
+        ):
+            self.do_path_regex_sub(
+                r'/var/lib/ovirt-engine/setup/answers/.*',
+                r'{key}=(.*)'.format(key=key),
+                r'{key}=********'.format(key=key)
+            )
+
 # vim: expandtab tabstop=4 shiftwidth=4
-- 
1.8.3.1

From 0a63fff9037848c8c240134fb8fff7e97e958a9d Mon Sep 17 00:00:00 2001
From: Sandro Bonazzola <sbonazzo redhat com>
Date: Tue, 16 Dec 2014 12:32:49 +0000
Subject: [PATCH] [ovirt] remove password leak from ovirt-engine setup answer
 file

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>
Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/ovirt.py | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/sos/plugins/ovirt.py b/sos/plugins/ovirt.py
index 407c918..6e3a092 100644
--- a/sos/plugins/ovirt.py
+++ b/sos/plugins/ovirt.py
@@ -186,4 +186,16 @@ class Ovirt(Plugin, RedHatPlugin):
                 r'{key}=********'.format(key=key)
             )
 
+        # Answer files contain passwords
+        for key in (
+            'OVESETUP_CONFIG/adminPassword',
+            'OVESETUP_CONFIG/remoteEngineHostRootPassword',
+        ):
+            self.do_path_regex_sub(
+                r'/var/lib/ovirt-engine/setup/answers/.*',
+                r'{key}=(.*)'.format(key=key),
+                r'{key}=********'.format(key=key)
+            )
+
+
 # vim: expandtab tabstop=4 shiftwidth=4
-- 
1.8.3.1

From da3f566dd23fc2a3a9a01b6679d1b646ea094dfc Mon Sep 17 00:00:00 2001
From: Tim Speetjens <tim.speetjens@redhat.com>
Date: Mon, 13 Jul 2015 11:41:24 +0200
Subject: [PATCH] [ovirt] obfuscate passwords in aaa profiles

With aaa profiles configured in /etc/ovirt-engine/aaa, all
.properties files in this directory may contain passwords in keys
vars.password, pool.default.ssl.truststore.password and
potentially pool.default.auth.simple.password

Resolves #588

Signed-off-by: Tim Speetjens <tim.speetjens@redhat.com>
---
 sos/plugins/ovirt.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sos/plugins/ovirt.py b/sos/plugins/ovirt.py
index 9436ac6..6f395a4 100644
--- a/sos/plugins/ovirt.py
+++ b/sos/plugins/ovirt.py
@@ -189,5 +189,15 @@ class Ovirt(Plugin, RedHatPlugin):
                 r'{key}=********'.format(key=key)
             )
 
+        # aaa profiles contain passwords
+        protect_keys = [
+            "vars.password",
+            "pool.default.auth.simple.password",
+            "pool.default.ssl.truststore.password"
+        ]
+        regexp = r"((?m)^\s*#*(%s)\s*=\s*)(.*)" % "|".join(protect_keys)
+
+        self.do_path_regex_sub("/etc/ovirt-engine/aaa/.*\.properties", regexp,
+                               r"\1*********")
 
 # vim: expandtab tabstop=4 shiftwidth=4
-- 
1.8.3.1

From 8b8c8bd1fe535038ff3372cd39413bb6765f1081 Mon Sep 17 00:00:00 2001
From: Nijin Ashok <nashok@redhat.com>
Date: Sat, 17 Oct 2015 19:36:20 +0530
Subject: [PATCH] [ovirt] Remove database password from AAA_JDBC profile files

Ovirt AAA_JDBC stores database password in plain text under
/etc/ovirt-engine/aaa/*.properties file.

It contains password in the format:

  config.datasource.dbpassword=<password>

The engine PostgreSQL user password and the custom build domain's  user
password exist in plain text format in these files. The commit removes
these password from the sosreport.

Fixes: #666.

Signed-off-by: Nijin Ashok <nashok@redhat.com>
---
 sos/plugins/ovirt.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sos/plugins/ovirt.py b/sos/plugins/ovirt.py
index 6f395a4..00f60d8 100644
--- a/sos/plugins/ovirt.py
+++ b/sos/plugins/ovirt.py
@@ -193,7 +193,8 @@ class Ovirt(Plugin, RedHatPlugin):
         protect_keys = [
             "vars.password",
             "pool.default.auth.simple.password",
-            "pool.default.ssl.truststore.password"
+            "pool.default.ssl.truststore.password",
+            "config.datasource.dbpassword"
         ]
         regexp = r"((?m)^\s*#*(%s)\s*=\s*)(.*)" % "|".join(protect_keys)
 
-- 
1.8.3.1

